### PR TITLE
[Integration]: GitLab Integration

### PR DESCRIPTION
**Pull request by senali-d**
## Description

Implement GitLab Handler for issues.

**Fixes** #5358

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)

### What is the solution?

GitLab Issues table for a given Repository (Support private and public repositories)

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation, or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+                    issues_kwargs["sort"] = an_order.field.parts[1]
+                    issues_kwargs["direction"] = an_order.direction
+                elif an_order.field.parts[1] in self.get_columns():
+                    order_by_conditions["columns"].append(an_order.field.parts[1])
+
+                    if an_order.direction == "ASC":
+                        order_by_conditions["ascending"].append(True)
+                    else:
+                        order_by_conditions["ascending"].append(False)
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py:58]: Can't find where `order_by_conditions` is used for applying sorting or conditions

File: mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        while start:
+            try:
+                issues = self.handler.connection.projects.get(
+                    self.handler.repository
+                ).issues.list()
+                for issue in issues:
+
+                    logger.debug(f"Processing issue {issue.iid}")
+
+                    gitlab_issues_df = pd.concat(
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py:105]: I think is better to have list of dicts (list of issues) and do pd.concat once after this circle. It should work faster.

**Review by ea-rus — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+        issues_kwargs = {}
+        order_by_conditions = {}
+
+        if query.order_by and len(query.order_by) > 0:
+            order_by_conditions["columns"] = []
+            order_by_conditions["ascending"] = []
+
+            for an_order in query.order_by:
+                if an_order.field.parts[0] != "issues":
```
[the remark this answers is not in the document]
Comment (senali-d) [reply mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py]: Apply order by condition to the select query

File: mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py (hunk 1 above)
Comment (senali-d) [reply mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py:47]: Apply order by condition to the select query

File: mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py (hunk 2 above)
Comment (senali-d) [reply mindsdb/integrations/handlers/gitlab_handler/gitlab_tables.py:96]: Issue list came with nested format. So keep as it is.

**Review by ZoranPandovski — APPROVED** [review]
Amazing, thanks @senali-d . Please make PR with other functionalities as MR

**Comment by senali-d** [comment]
@ZoranPandovski Sure.

**Pull request merged**